src/orion/algo/space.py

```diff
@@ -533,6 +538,11 @@ def get_prior_string(self):
         prior_string = super(Integer, self).get_prior_string()
         return prior_string[:-1] + ', discrete=True)'
 
+    @property
+    def prior_name(self):
+        """Return the name of the prior"""
+        return "int_uniform"
```

Comment:
The discrete dimension can have different priors. It should be `'int_{}'.format(super(Integer, self).prior_name)`

Reply:
The intention of this was more like to say no matter what prior specified for integer, tpe will always assume it is uniform. But I think it should be better to go with your suggestion, except tpe will only accept int_uniform. Will add this into the documentation when I have discrete and categorical support done.

Reply:
Yes, I understand why we need this, but the TPE will be fooled in thinking that the integer prior was uniform while the user is asking something else. Using the formatted prior name will lead to intended value error in TPE instantiation.